code/modules/projectiles/updated_projectiles/projectile.dm

```diff
@@ -321,18 +321,17 @@
 	if(!throwpass)
 		return TRUE
 
-	if(P.ammo.flags_ammo_behavior & AMMO_SNIPER) //sniper rounds bypass cover
+	if(P.ammo.flags_ammo_behavior & AMMO_SNIPER || P.ammo.flags_ammo_behavior & AMMO_SKIPS_HUMANS || P.ammo.flags_ammo_behavior & AMMO_ROCKET) //sniper, rockets and IFF rounds bypass cover
 		return FALSE
 
 	if(!(flags_atom & ON_BORDER))
 		return FALSE //window frames, unflipped tables
 
 	if(!( P.dir & reverse_direction(dir) || P.dir & dir))
 		return FALSE //no effect if bullet direction is perpendicular to barricade
-
-	var/distance = P.distance_travelled - 1
-
-	if(distance < 1)
+		
+	var/distance = P.distance_travelled
```

Comment:
This change will have an effect in `var/hitchance`. Are you taking it into account?
You could also delete this unnecessary var if you substituted `distance` with `P.distance_travelled` down there. Just note that before your changes that was `P.distance_travelled - 1`